android-database-sqlcipher/proguard-project.txt

```diff
@@ -0,0 +1,26 @@
+-optimizationpasses 5
+-dontusemixedcaseclassnames
+-dontskipnonpubliclibraryclasses
+-dontskipnonpubliclibraryclassmembers
+-dontpreverify
+-verbose
+-optimizations !code/simplification/arithmetic,!field/*,!class/merging/*,!code/allocation/variable
+
+-dontwarn javax.annotation.**
+
+-keepclasseswithmembernames class * {
+    native <methods>;
+}
+
+-keepclassmembers enum * {
+    public static **[] values();
+    public static ** valueOf(java.lang.String);
+}
+
+-keep class * implements android.os.Parcelable {
+  public static final android.os.Parcelable$Creator *;
+}
+
+-keep class net.sqlcipher.** {
```

Comment:
I think this line is keeping too much. This should be reduced to specific classes/methods/variables.
The reason for supplying Proguard rules in a library is for the clients (developers) to not waste time on examining the library's internals. However, this rule is useless as every developer can come up with such a solution for every library.